docs/installation/application.rst

```diff
@@ -0,0 +1,196 @@
+************************
+Application Installation
+************************
+
+**Last Updated:** March 13, 2019
+
+Once you have the portal configured and setup with all the required services, the next step is to install Tethys applications on to your portal.
+
+1. Navigate to App Directory
+============================
+
+Within the command line interface, navigate to the directory of the app you wish to install (this directory must contain the app's ``setup.py`` file).
+
+::
+
+    $ cd /directory/of/app
+
+2. Install the App
+==================
+
+Using Install (Recommended):
+----------------------------
+
+Using the ``install`` cli command, you can install the application in one quick step. The ``install`` command will install all the dependencies, link any services that you might need for your application and can also rely on the portal configuration to link default services to your application (as configured by the portal administrator).
+
+::
+
+    # Run Init
+    $ tethys install
+
+    # Tethys install with custom options
+    $ tethys install -f ../install.yml -p $TETHYS_HOME/src/configs/portal.yml
+
+
+The install command uses three configuration files:
+
+.. _tethys_install_yml:
+
+install.yml 
+-----------
+
+This file is generated with your application scaffold. Please do NOT list any dependencies in setup.py. Instead list them in the :file:`install.yml` file. This file should be committed with your application code in order to aid installation on a Tethys Portal
```

Comment:
What happens if the dependencies are listed in `setup.py`? If the dependencies are already installed with from the yaml file into the conda environment then it shouldn't matter if they are also listed here right?

Reply:
Based on a few tests that I ran installing apps with dependencies in setup.py or install.yml or both it doesn't seem to matter (its is probably better to list them in install.yml because you can also add the conda channel you want to search in). I don't think it is bad if we advise people to use a certain convention.  If you want me to rephrase it to be less 'harsh' I can. Instead of 'do NOT' I can say something like.. the preferred method is to add dependencies in install.yml not setup.py.  Let me know.

Reply:
@sdc50 I think this was meant to be more of a suggestion, but it could be rephrased. I don't think it will necessarily break anything to have dependencies listed in the setup.py. There is a pip section in the install files if you prefer to have a dependency installed with pip.

@tylorbayer, @sdc50 How about this wording: "...scaffold. *List any dependencies your app requires in this file instead of listing them in the setup.py.* This file should..."

Reply:
I guess I'm thinking about the possibility of having an app that is released as it's own package. In that case the dependencies should be listed in the `setup.py`. Also, one way (my preferred way) of creating a conda package is to extract the dependencies from the `setup.py` for the conda recipe. In any case I think we should be clear about what happens with the dependencies listed in either place. I recommend wording it something like this:

> Dependencies that are listed in the `install.yml` will be installed with conda and will honor the channel priority that is specified in the `install.yml`. If there are any dependencies listed in the `setup.py` that are not specified in the `install.yml` then these packages will be installed with pip as part of the setup process.

Reply:
@tylorbayer - please add @sdc50's suggested wording (see above) to clarify what happens when dependencies are listed in the setup.py. We should also remove the sentence: "Please do NOT list any dependencies in setup.py." It won't hurt if the dependencies are listed there.

Reply:
Resolved